scripts/conversion_script.py

```diff
@@ -0,0 +1,492 @@
+
+    new_message["key"] = "%s::%s" % (action_id, message["id"])
+    del new_message["id"]
+
+    return new_message
+
+
+def _generate_detail_block(message):
+    """
+    Helper method to generate the detail key that is composed by the
+    remediations and diagnosis fields.
+    Returns modified copy of original message.
+    """
+    new_message = copy.deepcopy(message)
+    detail_block = {
+        "remediations": [],
+        "diagnosis": [],
+    }
+
+    remediation_key = "remediations" if "remediations" in new_message else "remediation"
+    detail_block["remediations"].append(
+        {"context": new_message.pop(remediation_key, "")}
+    )
+    detail_block["diagnosis"].append({"context": new_message.pop("diagnosis", "")})
+    new_message["detail"] = detail_block
+    return new_message
+
+
+def _rename_dictionary_key(message, new_key, old_key):
+    """Helper method to rename keys in a flatten dictionary."""
+    new_message = copy.deepcopy(message)
+    new_message[new_key] = new_message.pop(old_key)
+    return new_message
+
+
+def _filter_message_level(message, level):
+    """
+    Filter for messages with specific level. If any of the message matches the
+    level, return None, otherwise, if it is different from what is expected,
+    return the message received to continue with the other transformations.
+    """
+    if message["level"] != level:
+        return message
+
+    return {}
+
+
+def apply_message_transform(message, action_id):
+    """Apply the necessary data transformation to the given messages."""
+    if not _filter_message_level(message, level="SUCCESS"):
+        return {}
+
+    new_message = _generate_message_key(message, action_id)
+    new_message = _rename_dictionary_key(new_message, "severity", "level")
+    new_message = _rename_dictionary_key(new_message, "summary", "description")
+    new_message = _generate_detail_block(new_message)
+
+    # Appending the `modifiers` key to the message here for now. Once we have
+    # this feature in the frontend, we can populate the data with it.
+    new_message["modifiers"] = []
+
+    return new_message
+
+
+def transform_raw_data(raw_data):
+    """
+    Method that will transform the raw data given and output in the expected
+    format.
+
+    The expected format will be a flattened version of both results and
+    messages into a single
+    """
+    new_data = []
+    for action_id, result in raw_data["actions"].items():
+        # Format the results as a single list
+        for message in result["messages"]:
+            new_data.append(apply_message_transform(message, action_id))
+
+        new_data.append(apply_message_transform(result["result"], action_id))
+
+    # Filter out None values before returning
+    return [data for data in new_data if data]
+
+
+def update_insights_inventory():
+    """Call insights-client to update insights inventory."""
+    output, returncode = run_subprocess(cmd=["/usr/bin/insights-client"])
```

Comment:
I am just thinking if we should add status print here as well (all other functions called in main have it)

```suggestion
    print('Updating system status in Red Hat Insights.')
    output, returncode = run_subprocess(cmd=["/usr/bin/insights-client"])
```